services/core/shared/core/compat/sdk_v5/accounts.js

```diff
@@ -43,20 +48,20 @@ const confirmPublicKey = async publicKey => {
 	return (account && account.publicKey === publicKey);
 };
 
-const getPublicKeyByAddress = async address => {
-	const accountsDB = await knex('accounts');
-	const [{ publicKey }] = await accountsDB.find({ address });
-	return publicKey;
-};
-
-const getIndexedAccountByPublicKey = async publicKey => {
-	const accountsDB = await knex('accounts');
-	const account = await accountsDB.find({ publicKey });
+const getIndexedAccountInfo = async params => {
+	const accountsDB = await getAccountsIndex();
+	const [account] = await accountsDB.find(params);
 	return account;
 };
 
-const getIndexedAccountInfo = async params => {
-	const accountsDB = await knex('accounts');
+const getAccountsBySearch = async searchString => {
+	const accountsDB = await getAccountsIndex();
+	const params = {
+		search: {
+			property: 'username',
+			pattern: searchString,
+		},
+	};
 	const account = await accountsDB.find(params);
 	return account;
 };
```

Comment:
Please consider renaming it to `getAccountsByUsernameSearch`.
```suggestion
const getAccountsByUsernameSearch = async searchString => {
	const accountsDB = await getAccountsIndex();
	const params = {
		search: {
			property: 'username',
			pattern: searchString,
		},
	};
	const account = await accountsDB.find(params);
	return account;
};
```

Or better make it more generic, with something similar to:

The `searchProp` and `searchString` can be wrapped into a single object as well.
```
{
    property: 'searchProp'
    pattern: 'searchString',
}
```
```suggestion
const getAccountsBySearch = async searchProp, searchString => {
	const accountsDB = await getAccountsIndex();
	const params = {
		search: {
			property: searchProp,
			pattern: searchString,
		},
	};
	const account = await accountsDB.find(params);
	return account;
};
```

Reply:
Done!